From ada6af7e6ffa3db33a4d04b0ddf466ea6722f30b Mon Sep 17 00:00:00 2001
From: Igor Grinberg <igrinberg@habana.ai>
Date: Tue, 09 Mar 2021 18:11:22 +0200
Subject: [PATCH] [SW-35518] gaudi: security: protect indirect msi

It was found that the indirect MSI mechanism is not working well in
various scenarios (e.g. during the card reset, or in the abnormal
interrupt storm situations, or host VM shutdown)
Protect the PCIe MSI region with RR so it can't be used as a security
threat.

Change-Id: I832b18994f5b932ae05639cc37120f59e2f59cea
Signed-off-by: Igor Grinberg <igrinberg@habana.ai>
---

diff --git a/gaudi/security.c b/gaudi/security.c
index 23bedcd..6e86dc6 100644
--- a/gaudi/security.c
+++ b/gaudi/security.c
@@ -1233,7 +1233,7 @@
 	LBW_RR_GICEND,
 	LBW_RR_PCIE_DBI_COREBGN,
 	LBW_RR_PCIE_COREEND_AUX_PHY,
-	LBW_RR_PSOC0,
+	LBW_RR_PCIE_MSI_PSOC0,
 	LBW_RR_PSOC1_NIC0_MAC_CH,
 	LBW_RR_NIC0_TS_NIC1_MAC_CH,
 	LBW_RR_NIC1_TS_NIC2_MAC_CH,
@@ -1295,10 +1295,10 @@
 	max = mmMMU_UP_BASE;
 	preboot_secure_lbw_rr(LBW_RR_PCIE_COREEND_AUX_PHY, min, max);
 
-	/* PSOC_I2C_M0 up to PSOC_TIMESTAMP: RR(LBW_RR_PSOC0) */
-	min = mmPSOC_I2C_M0_BASE - 1;
+	/* PCIE_MSI up to PSOC_TIMESTAMP: RR(LBW_RR_PCIE_MSI_PSOC0) */
+	min = mmPCIE_MSI_BASE - 1;
 	max = mmPSOC_TIMESTAMP_BASE;
-	preboot_secure_lbw_rr(LBW_RR_PSOC0, min, max);
+	preboot_secure_lbw_rr(LBW_RR_PCIE_MSI_PSOC0, min, max);
 
 	/* PSOC_GPIO0 up to NIC0_STAT: RR(LBW_RR_PSOC1_NIC0_MAC_CH) */
 	min = mmPSOC_GPIO0_BASE - 1;
